include/openenclave/host.h

```diff
@@ -156,6 +156,55 @@ oe_result_t oe_create_enclave(
     uint32_t ocall_count,
     oe_enclave_t** enclave);
 
+/**
+ * Create an enclave from an enclave image file with extended enclave
+ * initialization data.
+ *
+ * This function creates an enclave from an enclave image file. On successful
+ * return, the enclave is fully initialized and ready to use.
+ *
+ * @param path The path of an enclave image file in ELF-64 format. This
+ * file must have been linked with the **oecore** library and signed by the
+ * **oesign** tool.
+ *
+ * @param type The type of enclave supported by the enclave image file.
+ *     - OE_ENCLAVE_TYPE_SGX - An SGX enclave
+ *
+ * @param flags These flags control how the enclave is run.
+ *     It is the bitwise OR of zero or more of the following flags
+ *     - OE_ENCLAVE_FLAG_SIMULATE - runs the enclave in simulation mode
+ *     - OE_ENCLAVE_FLAG_DEBUG - runs the enclave in debug mode.
+ *                               DO NOT SHIP CODE with this flag
+ *
+ * @param settings Array of additional enclave creation settings for the
+ * specific enclave type.
+ *
+ * @param setting_count The number of settings in the **settings**.
+ *
+ * @param ocall_table Pointer to table of ocall functions generated by
+ * oeedger8r.
+ *
+ * @param ocall_count The number of functions in the **ocall_table**.
+ *
+ * @param eeid Optional user data for attestation.
+ *
+ * @param enclave This points to the enclave instance upon success.
+ *
+ * @returns Returns OE_OK on success.
+ *
+ */
+struct oe_eeid_t_;
+oe_result_t oe_create_enclave_eeid(
```

Comment:
Do we really need two different create_enclave APIs?

Reply:
No; as an alternative we can add the additional argument to the existing create_enclave, or perhaps put it into the settings. I figured it's best to keep it separate for now.